matlab/JSBSimInterface.cpp

```diff
@@ -0,0 +1,973 @@
+//     }
+    value = fdmExec->GetFCS()->GetDaRPos();
+    return 1;
+  }
+  else if (prop == "rudder-pos-rad")
+  {
+//     if ( verbosityLevel == eVeryVerbose )
+//     {
+//     	mexPrintf("\tEasy-set: rudder pos (deg) = %f\n",fdmExec->GetFCS()->GetDrPos());
+//     }
+    value = fdmExec->GetFCS()->GetDrPos();
+    return 1;
+  }
+  return 0;
+}
+//%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+bool JSBSimInterface::QueryJSBSimProperty(const string& prop)
+{
+	// mexPrintf("catalog size: %d\n",catalog.size());
+	for (unsigned i=0; i<catalog.size(); i++)
+	{
+		//mexPrintf("__%s__\n",catalog[i].c_str());
+		if (catalog[i].find(prop) != std::string::npos) return 1;
+	}
+	return 0;
+}
+//%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+void JSBSimInterface::PrintCatalog()
+{
+    const std::string& name = fdmExec->GetAircraft()->GetAircraftName();
+    mexPrintf("-- Property catalog for current aircraft ('%s'):\n", name.c_str());
+    for (unsigned i=0; i<catalog.size(); i++)
+        mexPrintf("%s\n",catalog[i].c_str());
+    mexPrintf("-- end of catalog\n");
+	return;
+}
+//%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+bool JSBSimInterface::Init(const mxArray *prhs1)
+{
+	// Inspired by "refbook.c"
+	// The argument prhs1 is pointer to a Matlab structure with two fields: name, value.
+	// The Matlab user is forced to build such a structure first, then to pass it to the
+	// mex-file. Example:
+	//    >> ic(1).name = 'u'; ic(1).value = 80; (ft/s)
+	//    >> ic(2).name = 'v'; ic(1).value =  0; (ft/s)
+	//    >> ic(3).name = 'w'; ic(1).value =  0; (ft/s)
+	//    >> ic(4).name = 'p'; ic(1).value =  0; (rad/s) % etc ...
+	//    >> MexJSBSim('init', ic)
+
+	//*************************************************
+	// Set dt=0 first
+
+	fdmExec->SuspendIntegration();
+	
+	//*************************************************
+
+	bool success = 1;
+
+	const char **fnames;       /* pointers to field names */
+	const mwSize *dims;
+	mxArray    *tmp;
+	char       *pdata=NULL;
+	int        ifield, nfields;
+	mxClassID  *classIDflags;
+	mwIndex    jstruct;
+	mwSize     NStructElems;
+	mwSize     ndim;
+
+    // get input arguments
+    nfields = mxGetNumberOfFields(prhs1);
+    NStructElems = mxGetNumberOfElements(prhs1);
+    // allocate memory  for storing classIDflags
+    classIDflags = (mxClassID*)mxCalloc(nfields, sizeof(mxClassID));
+    //mexPrintf("Very Verbose: %f", eVeryVerbose); 
+
+    // check empty field, proper data type, and data type consistency;
+	// and get classID for each field (see "refbook.c")
+    for(ifield=0; ifield<nfields; ifield++) 
+	{
+		for(jstruct = 0; jstruct < NStructElems; jstruct++) 
+		{
+			tmp = mxGetFieldByNumber(prhs1, jstruct, ifield);
+			if(tmp == NULL) 
+			{
+				return 0;
+			} 
+			if(jstruct==0) 
+			{
+				if( (!mxIsChar(tmp) && !mxIsNumeric(tmp)) || mxIsSparse(tmp)) 
+				{
+					return 0;
```

Comment:
```suggestion
					return false;
```
C++ uses `true` and `false` for booleans.